proto/common_definitions.proto

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" B1IS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+syntax = "proto3";
+
+package openconfig.attestz.proto;
+
+option go_package = "github.com/openconfig/attestz";
+
+enum ControlCardRole {
+  CONTROL_CARD_ROLE_UNSPECIFIED = 0;
+  CONTROL_CARD_ROLE_ACTIVE = 1;
+  CONTROL_CARD_ROLE_STANDBY = 2;
+}
+
+// Full description of a control card with switch vendor identity fields.
+message ControlCardVendorId {
+  ControlCardRole control_card_role = 1;
+  string control_card_serial = 2;
+  string control_card_slot = 3;
+  string chassis_manufacturer = 4;
+  string chassis_part_number = 5;
+  string chassis_serial_number = 6;
+}
+
+// Used for selecting a switch control card by one of its indentifiers.
+message ControlCardSelection {
+  oneof control_card_id {
+    ControlCardRole role = 1;
```

Comment:
to be clear, `control_card_id` is `oneof`, so you would only specify/select a single way to identify a control card. We would specify `ControlCardSelection` in the request when we either want to enroll or attest the device. The nice thing about using the role as opposed to control card serial in the request, is that we (1) don't have to track which control card serials map to which switch chassis and (2) we can handle DevID cert logic more gracefully: in our case TLS connection would be secured by IDevID cert of the primary control card for both primary and secondary RPC calls, which also means that for secondary control card its IDevID cert will be returned to us as part of the payload response.

Reply:
i am not great on the role concept as a selector as what woudl this look like for any system which has >2 control cards?

Reply:
I have not seen such devices with >2 control cards yet. What is the target use case for having more than 2?

All four vendors that we are engaged with at the moment have two- or one- control card devices. I am also fine with renaming the enum and values to something more generic (FIRST/SECOND, for example) that will accommodate the addition of another value. To be honest though, I would say if the use case for devices >2 control cards is not clear/inexistent, then we should not spend much time trying to accommodate such use case. We can always add another value to the enum or we can add another `oneof` field to `ControlCardSelection.control_card_id`. 

Let me know what you think and please include renaming suggestions if that you prefer that option.

Reply:
my issue is we have been discussing with vendors the ability to move from a 1/1 redundancy to a n+1 model so we could actually support clustering on chassis
I don't want to build in something here that is very orthogonal to that.  I don't see this really helping the workflow side of things - the workflow can easily see the state of the cards it needs to enroll from gnmi

Reply:
thanks for the context, Marcus. How do you suggest selecting a given control card in the case of n+1 model? Should we just assume a serial number? 

I am not very familiar with gnmi, but I would assume that those APIs may not be available yet at the time of first install when we actually do TPM enrollment? If that is not the case and the device has all necessary credentials/certs, please point me to the gnmi APIs that you think would be useful here.

There are a few reasons why using enum like PRIMARY vs SECONDARY is useful. First, we don't have to know which chassis or device hostname is tied to which control card serial. This is especially useful as we start looking into more complex use cases such as control card hot-swap. Another simplification is that it is a bit easier to deal with primary vs all other control cards when it comes to DevID handling. Since all communication is effectively done through primary control card, we will only be using primary's DevID for TLS, while the DevIDs of other control cards will be provided in the response payload. 

This is also something that one of the vendors has raised: b/285110110, which I think makes sense. Do you think it will be handy to keep this enum as one of the methods of selection at least for two-sup model? When we get to n+1 model we can either introduce another enum, expand existing ones or rely on serial numbers. If you agree I can rename the enum values to `CONTROL_CARD_ROLE_MODULAR_TWO_ACTIVE` and `CONTROL_CARD_ROLE_MODULAR_TWO_STANDBY`. I am open to better names :)